### DISCUSSION TITLE
Is there a 'Clean Old' function i a missing?

### DISCUSSION THREAD

**Discussion by MellyDawgKY** [rfennell/AzurePipelines#1371] [Q&A]
This product works great,  thank you so much for sharing, .....but  ( LOL!).  Does it have a clean option?  

First and foremost,  this product really does work great out of the box.  thank you so much for sharing publicly.  

I could not find mention of one, wrote a long tedious shell to parse wiki and circle back with MS wiki delete commands shared in api documentation by MS, only to hit a roadblock there with wiki type on the delete command, so I am back here scratching my head thinking there has to be a way to delete old wiki entries. 

every release I have it set to spit out a new document,  test and qa I want to limit as its growing a bit fast in size,  but every month i want to sweep it and leave last two.  Just not seeing anything like that unfortunately.  Any help would be awesome if not in the tool.

current configuration of ado:

- [ ] provisioned wiki within ado project
- [ ] yml build type
- [ ] classic release  type ( which generates docs)
- [ ] in classic release, use the two task by your company, your documentation task, and wiki update task is triggered through a deployment JOB step at the end of the deployment separately from the deployment agents on the boxes, so it stays in the cloud as much as possible

**Comment by rfennell** [comment]
You could use my wiki updater task, this will take the output of my release notes (or any other tool) and add or replace a page in a wiki.

I have no task to delete wiki pages, but you can do that as you said with a script as you suggest

**Comment by MellyDawgKY** [comment]
attaching a copy of what i came up with,  so far works like a charm, feel free to modernize and use as you see fit for your use.
[delete_wiki_shell.txt](https://github.com/rfennell/AzurePipelines/files/9964444/delete_wiki_shell.txt)
